jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -647,6 +647,124 @@ ecma_builtin_typedarray_prototype_reverse (ecma_value_t this_arg) /**< this argu
   return ecma_copy_value (this_arg);
 } /* ecma_builtin_typedarray_prototype_reverse */
 
+/**
+ * The %TypedArray%.prototype object's 'set' routine for a typedArray source
+ *
+ * See also:
+ *          ES2015, 22.2.3.22, 22.2.3.22.2
+ *
+ * @return ecma value of undefined.
+ */
+static ecma_value_t
+ecma_op_typedarray_set_with_typedarray (ecma_value_t this_arg, /**< this argument */
+                                        ecma_value_t arr_val, /**< typedarray object */
+                                        ecma_value_t offset_val) /**< offset value */
+{
+  /* 6.~ 8. targetOffset */
+  ecma_value_t ret_val = ECMA_VALUE_EMPTY;
+  ECMA_OP_TO_NUMBER_TRY_CATCH (target_offset_num, offset_val, ret_val);
+
+  if (ecma_number_is_nan (target_offset_num))
+  {
+    target_offset_num = 0;
+  }
+
+  if (target_offset_num <= -1.0 || target_offset_num >= (ecma_number_t) UINT32_MAX + 0.5)
+  {
+    return ecma_raise_range_error (ECMA_ERR_MSG ("Invalid offset"));
```

Comment:
Normally you cannot return from an ECMA_OP_TO_NUMBER_TRY_CATCH (or any similar macros). But here it would be better just not use this macro, so you could return anytime.